src/source/CppGenerator.scala

```diff
@@ -54,11 +54,12 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
     val self = marshal.typename(ident, e)
 
     if (spec.cppEnumHashWorkaround) {
+      refs.hpp.add("#include <type_traits>") // needed for std::underlying_type
       refs.hpp.add("#include <functional>") // needed for std::hash
     }
 
     writeHppFile(ident, origin, refs.hpp, refs.hppFwds, w => {
-      w.w(s"enum class $self : int").bracedSemi {
+      w.w(s"enum class $self : unsigned int").bracedSemi {
```

Comment:
I was pretty sure that the response would have been that one. :-)
I'm going to rollback it with the next commit. It makes perfectly sense.